### PR TITLE
refactor(core/presentation): Change how ReactModal works

### PR DESCRIPTION
**Pull request by christopherthielen**
- Render the Modal component automatically
- Pass a Component class, instead of a bunch of JSX.Element
- Provide closeModel/dismissModal callbacks to the contents component

### CODE REVIEW

**Comment by jrsquared** [comment]
🥇

File: app/scripts/modules/core/src/presentation/ModalService.tsx [hunk 1]
```diff
@@ -1,11 +1,52 @@
...
+   * @param props to pass to the Modal and ModalComponent
+   * @returns {Promise<T>}
+   */
+  public static show<P extends IModalComponentProps, T = any>(ModalComponent: React.ComponentType<P>, props?: P): Promise<T> {
+    const modalPropKeys: [keyof ModalProps] = [
+      'onHide', 'animation', 'autoFocus', 'backdrop', 'backdropClassName', 'backdropStyle',
+      'backdropTransitionTimeout', 'bsSize', 'container', 'containerClassName', 'dialogClassName',
+      'dialogComponent', 'dialogTransitionTimeout', 'enforceFocus', 'keyboard', 'onBackdropClick',
+      'onEnter', 'onEntered', 'onEntering', 'onEscapeKeyUp', 'onExit', 'onExited', 'onExiting',
+      'onShow', 'transition',
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/presentation/ModalService.tsx]: I like how this list is mostly alphabetical, but then you throw `onHide` at the beginning to keep us on our toes.

Comment (christopherthielen) [reply app/scripts/modules/core/src/presentation/ModalService.tsx]: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/38a16c7ac7a23708e5bcb0b40239ccf08ac136bb/types/react-bootstrap/lib/Modal.d.ts#L10-L34

Comment (anotherchrisberry) [reply app/scripts/modules/core/src/presentation/ModalService.tsx]: I liked my story better

**Pull request merged**